var/spack/repos/builtin/packages/amp/package.py

```diff
@@ -0,0 +1,145 @@
+    homepage = "https://bitbucket.org/AdvancedMultiPhysics/amp"
+
+    # ###################### Versions ##########################
+    #version("1.1", hg="https://bitbucket.org/AdvancedMultiPhysics/amp", tag="v1.1")
+    version("develop", hg="https://bitbucket.org/AdvancedMultiPhysics/amp")
+
+    # ###################### Variants ##########################
+
+
+
+    # ###################### Dependencies ##########################
+
+    # Everything should be compiled position independent (-fpic)
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('boost', when='+boost')
+    depends_on('petsc', when='+petsc')
+    depends_on('trilinos', when='+trilinos')
+    depends_on('hdf5', when='+hdf5')
+    depends_on('hdf5', when='+silo')
+    depends_on('silo', when='+silo')
+    depends_on('zlib', when="+zlib")
+
+    # MPI related dependencies
+    depends_on('mpi', when='+mpi')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        cxx_flags = []
+        options = []
+
+
+        # #################### Base Settings #######################
+
+        options.extend([
+            '-DTPL_URL=https://bitbucket.org/AdvancedMultiPhysics/tpl-builder',
+            '-DAMP_DATA_URL=https://bitbucket.org/AdvancedMultiPhysics/amp/downloads/AMP-Data.tar.gz',
+            '-DAMP_ENABLE_TESTS:BOOL=OFF',
+            '-DAMP_ENABLE_EXAMPLES:BOOL=OFF',
+            '-DAMP_ENABLE_CXX11:BOOL=ON',
+            '-DCXX_STD=11',
+            '-DBUILD_SHARED_LIBS:BOOL=%s' % (
+                'ON' if '+shared' in spec else 'OFF'),
+        ])
+        #    -D CMAKE_BUILD_TYPE=Debug
+
+        # #################### Compiler Settings #######################
+
+        if '+mpi' in spec:
+            options.extend([
+                '-DCMAKE_C_COMPILER=%s'       % spec['mpi'].mpicc,
+                '-DCMAKE_CXX_COMPILER=%s'     % spec['mpi'].mpicxx,
+                '-DCMAKE_Fortran_COMPILER=%s' % spec['mpi'].mpifc,
+                '-DUSE_MPI=0',
+                '-DMPI_COMPILER=1',
+                '-DMPIEXEC=$s'                % spec['mpi'].prefix.bin,
+            ])
+        else:
+            options.extend([
+                '-DCMAKE_C_COMPILER=%s'       % self.compiler.cc,
+                '-DCMAKE_CXX_COMPILER=%s'     % self.compiler.cxx,
+                '-DCMAKE_Fortran_COMPILER=%s' % self.compiler.fc,
+                '-DUSE_MPI=0',
+            ])
+        # -D FFLAGS="-fPIC"                               \
+        # -D CFLAGS="-fPIC"                               \
+        # -D CXXFLAGS="-fPIC"                             \
+        # -D LDFLAGS=""
+
+
+        # ################## Third Party Libraries #####################
+
+        tpl_list = "LAPACK"
+        options.extend([ '-DTPL_LAPACK_INSTALL_DIR=%s' % spec['lapack'].prefix, ])
```

Comment:
can you explicitly set blas/lapack libraries? see http://spack.readthedocs.io/en/latest/packaging_guide.html#blas-lapack-and-scalapack-libraries